### DISCUSSION TITLE
Reading values from a non-input component.

### DISCUSSION THREAD

**Discussion by T41WO** [react-hook-form/react-hook-form#3915] [Q&A] [answered]
I have a custom component (list of tags/buttons) in the midst of my form. I would like to have add this component values, i.e. array of the list tags/button texts, to my form data. 

See image below, blue tags with NBA team names. 
![image](https://user-images.githubusercontent.com/7270970/104796359-4aeb0c00-576a-11eb-921a-49767c9152ed.png)

I wrapped the TagList with the controller
![image](https://user-images.githubusercontent.com/7270970/104796425-ed0af400-576a-11eb-9231-5a8bd823c987.png)

How do I return "tags" to the parent form and have it available in getValues and/or formData when submitting or validating?

**Comment by bluebill1049** [comment, marked as answer]
I think you should use `useFieldArray`, for the array field section.

**Reply by T41WO** [reply]
Thanks. This helped. Made Progress
